docs/advanced-guides/oauth-client.md

```diff
@@ -0,0 +1,283 @@
+---
+sidebar_position: 17
+---
+
+## OAuth Client Implementation
+
+This is a guide to implementing atproto OAuth clients "The Hard Way". Optimistically, most developers will have an SDK available for their programming language which supports OAuth, and they can simply refer to SDK documentation. This guide is intended for early adopters, SDK maintainers, or developers with more sophisticated OAuth needs. It is agnostic to whether developers are building clients to work the the `app.bsky` microblogging Lexicons, or implementing novel application Lexicons.
+
+The [atproto OAuth specification](https://github.com/bluesky-social/atproto-website/pull/326) is the authoritative document on how to use OAuth with atproto. If there are discrepancies between this document and the specification, defer to the specification. This document skips over a few details and uses more atproto-specific terminology, but readers are still expected to be familiar with OAuth 2 concepts, terminology, and standards.
```

Comment:
```suggestion
The [atproto OAuth specification](https://github.com/bluesky-social/atproto-website/pull/326) is the authoritative document on how to use OAuth with atproto. If there are discrepancies between this document and the specification, defer to the specification. This document skips over a few details and uses more atproto-specific terminology, but readers are still expected to be familiar with OAuth 2.1 concepts, terminology, and standards.
```

Reply:
I had this written as "OAuth 2.0/2.1" and shortened it to "OAuth 2" instead. For general devs (the target of this document), i'm worried that referencing 2.1 will be a rabbit hole? It is till a draft, not a standard/RFC yet, and I bet a lot of folks aren't familiar.

I don't have a strong opinion though, I think either/both would be fine, it is a bit subtle.